src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2937,6 +2937,42 @@ SqlTaskList(Job *job)
 }
 
 
+/*
+ * RelabelTypeMutator walks over the tree and converts RelabelType's
+ * into CollationExpr's. With that, we will be able to pushdown COLLATE's.
+ */
+Node *
+RelabelTypeMutator(Node *originalNode)
+{
+	if (originalNode == NULL)
+	{
+		return NULL;
+	}
+
+	if (IsA(originalNode, RelabelType))
+	{
+		RelabelType *originalRelabelType = (RelabelType *) originalNode;
+		if (OidIsValid(originalRelabelType->resultcollid))
+		{
+			CollateExpr *newCollateExpr = makeNode(CollateExpr);
+			newCollateExpr->arg = originalRelabelType->arg;
+			newCollateExpr->collOid = originalRelabelType->resultcollid;
+			newCollateExpr->location = originalRelabelType->location;
+			return expression_tree_mutator((Node *) newCollateExpr,
+										   RelabelTypeMutator,
+										   (void *) NULL);
+		}
+	}
+	else if (IsA(originalNode, Query))
+	{
+		return (Node *) query_tree_mutator((Query *) originalNode, RelabelTypeMutator,
+										   (void *) NULL, 0);
+	}
+	return expression_tree_mutator(originalNode, RelabelTypeMutator,
```

Comment:
since the nodes will be visited recursively by the deparsing logic, do we still need a mutator?